### PR TITLE
[CHORE tests] modernize unit/model/lifecycle-callbacks-test

### PR DESCRIPTION
**Pull request by runspired**
ported from #5708

### CODE REVIEW

File: tests/unit/model/lifecycle-callbacks-test.js [hunk 1]
```diff
@@ -1,299 +1,1118 @@
...
+    lifecycleEventMethodCalls = 0;
 
-  const Person = DS.Model.extend({
-    bar: DS.attr('string'),
-    name: DS.attr('string'),
+    await settled();
+    assert.strictEqual(
+      lifecycleEventMethodCalls,
+      1,
+      'We do not trigger didDelete when we first call record.deleteRecord'
```
Comment (igorT) [inline tests/unit/model/lifecycle-callbacks-test.js]: We *do* trigger didDelete

File: tests/unit/model/lifecycle-callbacks-test.js [hunk 2]
```diff
@@ -1,299 +1,1118 @@
...
+      assert.strictEqual(lifecycleEventMethodCalls, 1, 'We trigger becameInvalid when we reject');
 
-          assert.equal(reason.errors.length, 1, 'reason should have one error');
-          assert.equal(reason.errors[0].title, 'Invalid Attribute');
-          assert.equal(callCount, 1, 'becameInvalid called after invalidating');
-        });
-      });
+      assert.ok(reason.isAdapterError, 'reason should have been an adapter error');
+      assert.equal(reason.errors.length, 1, 'reason should have one error');
+      assert.equal(reason.errors[0].title, 'Invalid Attribute');
```
Comment (igorT) [inline tests/unit/model/lifecycle-callbacks-test.js]: We're missing an assertion here for triggering didUpdate

File: tests/unit/model/lifecycle-callbacks-test.js [hunk 3]
```diff
@@ -1,299 +1,1118 @@
...
-  // Make sure that the error handler has a chance to attach before
-  // save fails.
-  return run(() => {
-    return asyncPerson.then(person => {
-      return run(() => {
-        person.set('bar', 'Bar');
-        return person.save().catch(reason => {
-          assert.ok(reason.isAdapterError, 'reason should have been an adapter error');
+    await person.save().catch(reason => {
+      assert.strictEqual(lifecycleEventMethodCalls, 1, 'We trigger becameInvalid when we reject');
```
Comment (igorT) [inline tests/unit/model/lifecycle-callbacks-test.js]: Just curious, what are the reasons for preferring `strictEqual` over `equal` in some cases?

File: tests/unit/model/lifecycle-callbacks-test.js (hunk 1 above)
Comment (igorT) [reply tests/unit/model/lifecycle-callbacks-test.js]: Fixed

File: tests/unit/model/lifecycle-callbacks-test.js (hunk 3 above)
Comment (runspired) [reply tests/unit/model/lifecycle-callbacks-test.js]: I use `strictEqual` when we do expect strict referential equality. The advantage is two-fold: (1) it avoids a set of bugs that happen when qunit attempts to do deep recursive "looks like the same thing" comparisons when strict equality is not matched (on records this is a mega perf hit in addition to being risky), and it is an improvement over the alternative pattern of `assert.ok(foo === bar, '')` wherein if the assertion fails there is no additional information printed about what the incorrect value is without stepping into the test.

File: tests/unit/model/lifecycle-callbacks-test.js (hunk 1 above)
Comment (runspired) [reply tests/unit/model/lifecycle-callbacks-test.js]: thanks!

**Pull request merged**